### PR TITLE
Add `no-absolute-path` rule (fixes #530)

### LINKED ISSUE #530: New rule: no-absolute-path
Based off [this comment](https://github.com/benmosher/eslint-plugin-import/issues/512#issuecomment-241761338)

This rule should forbid the use of imports with an absolute path. This is valid in Node.js, but should not appear in code in npm packages or in production, or better: never.

### Invalid

```js
import Foo from '/some/absolute/path.js';
```

### Valid

```js
import Foo from './some/relative/path.js';
import Foo from 'foo';
```

### PR DESCRIPTION
**Pull request by jfmengels**
Add `no-absolute-path` rule (fixes #530)

### CODE REVIEW

File: tests/src/core/importType.js [hunk 1]
```diff
@@ -8,6 +8,12 @@ import { testContext } from '../utils'
 describe('importType(name)', function () {
   const context = testContext()
 
+  it("should return 'absolute' for node.js modules", function() {
```
Comment (benmosher) [inline tests/src/core/importType.js]: What does 'node.js' mean here?

**Comment by benmosher** [comment]
LGTM! :shipit:

File: tests/src/core/importType.js (hunk 1 above)
Comment (jfmengels) [inline tests/src/core/importType.js]: That means typo :sweat_smile:

Comment (jfmengels) [inline tests/src/core/importType.js]: Fixed

**Pull request merged**